packages/client/src/runtime/core/types/Utils.ts

```diff
@@ -118,6 +109,9 @@ export type PayloadToResult<P, O extends Record<any, any> = RenameAndNestPayload
 
 export type Select<T, U> = T extends U ? T : never
 
+// prettier-ignore
+export type Equals<A, B> = (<T>() => T extends A ? 1 : 2) extends (<T>() => T extends B ? 1 : 2) ? 1 : 0
```

Comment:
Could you explain what this type is doing and add this in a comment? I have really hard time grasping it.

Reply:
It comes from another PR see discussion https://github.com/prisma/prisma/pull/20180#discussion_r1270060230

Reply:
It comes from a former Microsoft employee
https://github.com/microsoft/TypeScript/issues/27024#issuecomment-421529650

Reply:
It abuses some of the variance rules in covariant positions to assert that a type is strictly equal to another.